### PR TITLE
Fix simple module

### PR DESCRIPTION
**Pull request by cheukt**
Validation shouldn't be setting variables on the class, and we generally shouldn't have instance-specific vars as class vars either.
Ran into this while investigating RSDK-4171

### CODE REVIEW

File: docs/examples/module_step2.py [hunk 1]
```diff
@@ -26,9 +26,6 @@ async def get_readings(self, extra: Optional[Dict[str, Any]] = None, **kwargs) -
         wifi_signal = [x for x in content[2].split(" ") if x != ""]
         return {"link": wifi_signal[2], "level": wifi_signal[3], "noise": wifi_signal[4]}
 
-    async def get_geometries(self, *, extra: Optional[Dict[str, Any]] = None, timeout: Optional[float] = None) -> List[Geometry]:
```
Comment (purplenicole730) [inline docs/examples/module_step2.py:29]: Does every `Sensor` need to define `get_geometries()` or is it unnecessary?

Comment (purplenicole730) [inline docs/examples/module_step2.py:29]: Does every `Sensor` need to define `get_geometries()` or is it unnecessary?

**Review by purplenicole730 — APPROVED** [review]
LGTM, although I did have questions about `get_geometries()`. Is it that not every `Sensor` needs it defined, or that the simple module was missing it?

File: docs/examples/module_step2.py (hunk 1 above)
Comment (cheukt) [reply docs/examples/module_step2.py:29]: it's unnecessary because the `ComponentBase` parent class already implements it as unimplemented

**Pull request merged**